client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/packages-step/index.js

```diff
@@ -0,0 +1,191 @@
+import { hasNonEmptyLeaves } from 'woocommerce/woocommerce-services/lib/utils/tree';
+import StepContainer from '../step-container';
+import {
+	getShippingLabel,
+	isLoaded,
+	getFormErrors,
+} from 'woocommerce/woocommerce-services/state/shipping-label/selectors';
+import { toggleStep, confirmPackages } from 'woocommerce/woocommerce-services/state/shipping-label/actions';
+
+const PackagesStep = ( props ) => {
+	const {
+		siteId,
+		orderId,
+		selected,
+		all,
+		weightUnit,
+		errors,
+		expanded,
+		translate,
+	} = props;
+
+	const packageIds = Object.keys( selected );
+	const itemsCount = packageIds.reduce( ( result, pId ) => ( result + selected[ pId ].items.length ), 0 );
+	const totalWeight = packageIds.reduce( ( result, pId ) => ( result + selected[ pId ].weight ), 0 );
+	const isValidWeight = packageIds.reduce( ( result, pId ) => ( result && 0 < selected[ pId ].weight ), true );
+	const isValidPackageType = packageIds.reduce( ( result, pId ) => ( result && 'not_selected' !== selected[ pId ].box_id ), true );
+	const isValidPackages = 0 < packageIds.length;
+	const hasAnyPackagesConfigured = all && Object.keys( all ).length;
+
+	const getContainerState = () => {
+		if ( ! isValidPackages ) {
+			return {
+				isError: true,
+				summary: translate( 'No packages selected' ),
+			};
+		}
+
+		if ( ! isValidPackageType ) {
+			return {
+				isError: true,
+				summary: translate( 'Please select a package type' ),
+			};
+		}
+
+		if ( ! isValidWeight ) {
+			return {
+				isError: true,
+				summary: translate( 'Weight not entered' ),
+			};
+		}
+
+		let summary = '';
+
+		if ( 1 === packageIds.length && 1 === itemsCount ) {
+			summary = translate( '1 item in 1 package: %(weight)f %(unit)s total', {
+				args: {
+					weight: totalWeight,
+					unit: weightUnit,
+				},
+			} );
+		} else if ( 1 === packageIds.length ) {
+			summary = translate( '%(itemsCount)d items in 1 package: %(weight)f %(unit)s total', {
+				args: {
+					itemsCount,
+					weight: totalWeight,
+					unit: weightUnit,
+				},
+			} );
+		} else {
+			summary = translate( '%(itemsCount)d items in %(packageCount)d packages: %(weight)f %(unit)s total', {
+				args: {
+					itemsCount,
+					packageCount: packageIds.length,
+					weight: totalWeight,
+					unit: weightUnit,
+				},
+			} );
+		}
+
+		if ( ! hasAnyPackagesConfigured ) {
+			return { isWarning: true, summary };
+		}
+
+		return { isSuccess: true, summary };
+	};
+
+	const renderWarning = () => {
+		if ( ! hasAnyPackagesConfigured ) {
+			const className = 'error-notice';
+
+			return <Notice
+				className={ className }
+				status="is-warning"
+				showDismiss={ false } >
+				<span>{ translate(
+					'There are no packages configured. The items have been packed individually. ' +
+					'You can add or enable packages using the {{a}}Packaging Manager{{/a}}.',
+					{ components: { a: <a href="admin.php?page=wc-settings&tab=shipping&section=package-settings" /> } }
```

Comment:
⚠️ WP-Admin link!

Reply:
If this is never going to happen (`props.all` always has properties, so `hasAnyPackagesConfigured` will always be true), then this `renderWarning()` and `props.all` should be removed entirely.